### PR TITLE
UCS/LINEAR_FUNC: Pass function struct by value for easier usage

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by dmitrygx** [comment]
unrelated failure:
```
[----------] 2 tests from malloc_hook_cplusplus, where TypeParam = 
[ RUN      ] malloc_hook_cplusplus.remap_override_single_thread <> <>
/scrap/azure/agent-09/AZP_WORKSPACE/2/s/contrib/../test/gtest/ucm/malloc_hook.cc:784: Failure
Expected: (m_mapped_size) >= ((mapped_size) + (size)), actual: 9830400 vs 13107200
single-thread
[swx-rdmz-ucx-gpu-01:16466:0:16466] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x1fffe23e8)
make: *** [test] Segmentation fault
```

**Comment by yosefe** [comment]
bot:pipe:retest

File: test/gtest/ucs/test_math.cc [hunk 1]
```diff
@@ -209,14 +209,14 @@ UCS_TEST_F(test_math, linear_func) {
...
-        y[i]      = ucs_linear_func_apply(&func[i], x);
+        func[i] = ucs_linear_func_make(ucs::rand() / (double)RAND_MAX,
+                                       ucs::rand() / (double)RAND_MAX);
+        y[i]    = ucs_linear_func_apply(func[i], x);
     }
 
     ucs_linear_func_t sum_func;
-    ucs_linear_func_add(&sum_func, &func[0], &func[1]);
-    double y_sum = ucs_linear_func_apply(&sum_func, x);
+    sum_func = ucs_linear_func_add(func[0], func[1]);
```
Comment (brminich) [inline test/gtest/ucs/test_math.cc:218]: minor: can align by `=`

**Pull request merged**